test/kube2e/gloo/gloo_resources_test.go

```diff
@@ -190,21 +195,22 @@ var _ = Describe("GlooResourcesTest", func() {
 					kubev1.TLSPrivateKeyKey: []byte(crtKey),
 				}
 				_, err := resourceClientset.KubeClients().CoreV1().Secrets(tlsSecret.GetNamespace()).Update(ctx, tlsSecret, metav1.UpdateOptions{})
-				Expect(err).NotTo(HaveOccurred())
+				g.Expect(err).NotTo(HaveOccurred())
 
 				By("Eventually can curl the endpoint")
-				testHelper.CurlEventuallyShouldRespond(helper.CurlOpts{
+				testHelper.CurlEventuallyShouldRespondWithGomega(g, helper.CurlOpts{
 					Protocol:          "http",
 					Path:              "/",
 					Method:            "GET",
-					Host:              helper.TestrunnerName,
+					Host:              helper.TestServerName,
 					Service:           defaults.GatewayProxyName,
-					ConnectionTimeout: 1,
+					Port:              gatewayPort,
+					ConnectionTimeout: int(timeForCurling / time.Second),
 					WithoutStats:      true,
 					Verbose:           true,
-				}, kube2e.GetSimpleTestRunnerHttpResponse(), 0, 60*time.Second, 1*time.Second)
-
-			}, timeInBetweenRotation*timesToPerform, timeInBetweenRotation).Should(Succeed())
+				}, kube2e.TestServerHttpResponse(), 1, rotationInterval, timeForCurling)
+			}, rotationInterval*time.Duration(numAttempts),
```

Comment:
```suggestion
			}, rotationInterval,
```

If I understand correctly, as currently written, we will allow each individual attempt the amount of time we expect all attempts to require in total

Reply:
The timeout on an Eventually is the total time to allow for the condition to be met. We want this total time on this outer Eventually to be the max time allowed, which is the `rotationInterval` (i.e. the maximum time for the inner Eventually) times `numAttempts` (the number of attempts we want to make at the inner Eventually).

Reply:
I removed this nested `Eventually` such that we assert any failure of the `CurlShouldEventuallyRespond` is unacceptable.